### ISSUE TITLE
Alarm widget under the clock randomly showing up

### ISSUE THREAD

**Issue by Lock-The-Door** [MM2-0/Kvaesitso#172]
For some reason the next alarm widget keeps showing up under the clock even though my alarm is not within the next 15 minutes.

**Comment by Lock-The-Door** [comment]
It appears to have some correlation with plugging in my phone, (it switches from the charging widget to the alarm widget after a few seconds)

**Comment by MM2-0** [comment]
Some calendar apps interfere with the alarm widget; calendar events will appear as alarms if they have reminders enabled. It's a known issue but I haven't found the time yet to investigate why this is happening and how to prevent this.

**Comment by Lock-The-Door** [comment]
After some digging, the AlarmManager class seems to be used by applications to schedule tasks. It appears that the only way to get a user's clocks is by querying the clock app's api which will unfortunately vary between users. https://stackoverflow.com/a/3147407

**Comment by MM2-0** [comment]
I suspected something like that. Thanks, I'll look into that.

> which will unfortunately vary between users

If it works on AOSP / LineageOS / Pixel Android, it's good enough for me.

**Comment by tzugen** [comment]
Yes, there are actually different APIs that apps should use to schedule Tasks (like WorkManager).
The AlarmManager app should only be used by apps actually using alarms.

I would say that this is not a bug of Kvaesito. Maybe somehow it could be made to try to query the stock Clock app for alarms, but then any other alarms, like from any non-stock alarm app would not show up anymore.

I would rather suggest to @Lock-The-Door to raise a bug with the app that misuses this API.

**Comment by MM2-0** [comment]
Google Calendar schedules event notification without it showing up in the calendar widget. Don't know how they do it but there seems to be a way.

What I wonder is, do these false alarms appear in the status bar @Lock-The-Door ?

Afaik, only alarms scheduled using [`AlarmManager.setAlarmClock`](https://developer.android.com/reference/android/app/AlarmManager.html#setAlarmClock(android.app.AlarmManager.AlarmClockInfo,%20android.app.PendingIntent)) should appear in the clock widget but using that method would also make the system show the alarm indicator in the status bar so something isn't right.

**Comment by Lock-The-Door** [comment]
> What I wonder is, do these false alarms appear in the status bar @Lock-The-Door ?

No they do not

**Comment by JetpackJackson** [comment]
I noticed the other day that a reminder I scheduled on the Google Calendar appeared in the clock area, I guess their reminder system uses that API.

**Comment by mtrue01** [comment]
This bug is affecting my launcher as, I do not have  any third-party task management software or calendar software installed and this bug has literally been present for years for me, it's why I disable the alarms widget

**Comment by HoneyPie20** [comment]
Hello, is there currently a solution to this problem?
